pkg/clusterregistry/options/options.go

```diff
@@ -26,31 +26,38 @@ import (
 	"github.com/spf13/pflag"
 )
 
-// Runtime options for the clusterregistry-apiserver.
+// Runtime options for the cluster registry.
 type ServerRunOptions struct {
-	GenericServerRunOptions *genericoptions.ServerRunOptions
-	Etcd                    *genericoptions.EtcdOptions
-	SecureServing           *genericoptions.SecureServingOptions
-	Audit                   *genericoptions.AuditOptions
-	Features                *genericoptions.FeatureOptions
-	Authentication          *BuiltInAuthenticationOptions
+	GenericServerRunOptions  *genericoptions.ServerRunOptions
+	Etcd                     *genericoptions.EtcdOptions
+	SecureServing            *genericoptions.SecureServingOptions
+	Audit                    *genericoptions.AuditOptions
+	Features                 *genericoptions.FeatureOptions
+	StandaloneAuthentication *StandaloneAuthenticationOptions
+	StandaloneAuthorization  *StandaloneAuthorizationOptions
+	DelegatingAuthentication *genericoptions.DelegatingAuthenticationOptions
+	DelegatingAuthorization  *genericoptions.DelegatingAuthorizationOptions
 
-	EventTTL time.Duration
+	EventTTL         time.Duration
+	UseDelegatedAuth bool
 }
 
 // NewServerRunOptions creates a new ServerRunOptions object with default values.
 func NewServerRunOptions() *ServerRunOptions {
 	o := &ServerRunOptions{
 		GenericServerRunOptions: genericoptions.NewServerRunOptions(),
-		Etcd:           genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
-		SecureServing:  genericoptions.NewSecureServingOptions(),
-		Audit:          genericoptions.NewAuditOptions(),
-		Features:       genericoptions.NewFeatureOptions(),
-		Authentication: NewBuiltInAuthenticationOptions().WithAll(),
+		Etcd:                     genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig("/registry/clusterregistry.kubernetes.io", nil)),
+		SecureServing:            genericoptions.NewSecureServingOptions(),
+		Audit:                    genericoptions.NewAuditOptions(),
+		Features:                 genericoptions.NewFeatureOptions(),
+		StandaloneAuthentication: NewStandaloneAuthenticationOptions().WithAll(),
+		StandaloneAuthorization:  NewStandaloneAuthorizationOptions(),
+		DelegatingAuthentication: genericoptions.NewDelegatingAuthenticationOptions(),
+		DelegatingAuthorization:  genericoptions.NewDelegatingAuthorizationOptions(),
 
-		EventTTL: 1 * time.Hour,
+		EventTTL:         1 * time.Hour,
+		UseDelegatedAuth: true,
```

Comment:
Do we want `true` to be the default? Is this consistent with the behavior elsewhere? I mean, do we default to delegated+aggregated mode in other places too?

Reply:
@font suggested that I default this to true, because we expect that most people will want to deploy an aggregated registry. In the end, if people use `crinit`, this flag will not be exposed to them, so I think we have some flexibility to change it later.

Reply:
Ok.